### PR TITLE
reconfigure testing suite to match climacore

### PR DESCRIPTION
**Pull request by nefrathenrici**
## Content
- Updates the test project and pipeline to match climacore format
- Remove old versions of thermodynamics, since they break the tests
- Increase the minor version because this is a breaking change

### CODE REVIEW

**Review by trontrytel — COMMENTED** [review]
LGTM! @akshaysridhar - could you also take a look?

**Review by trontrytel — APPROVED** [review]

File: test/runtests.jl [hunk 1]
```diff
@@ -20,18 +20,18 @@ thermo_params = SFP.thermodynamics_params(param_set)
 uft = SFP.universal_func_type(param_set)
 
 const TD = Thermodynamics
+device(::T) where {T <: Array} = CPU()
```
Comment (akshaysridhar) [inline test/runtests.jl:23]: Thanks @nefrathenrici This will get an overhaul in the next couple of weeks as we start to move towards enabling GPU tests.

**Comment by akshaysridhar** [comment]
Changes LGTM. Is the idea to limit the compat entry in `ClimaAtmos` for this package to `v0.6` for the time being ? As we upgrade the tests + docs / add roughness-sublayer models etc. we are likely to have additional breaking changes.

**Review by akshaysridhar — APPROVED** [review]

**Pull request merged**